oracle/src/main.rs

```diff
@@ -21,23 +21,77 @@ const NAME: &str = env!("CARGO_PKG_NAME");
 const ABOUT: &str = env!("CARGO_PKG_DESCRIPTION");
 
 const CONFIRMATION_TARGET: u32 = 1;
+
 const BTC_DECIMALS: u32 = 8;
-const DOT_DECIMALS: u32 = 10;
+const BTC_CURRENCY: &str = "btc";
+
+macro_rules! create_collateral_id {
```

Comment:
Can't we reuse the primitives crate?

Reply:
We could but there are two issues:
- `CollateralId`is a subset
- We may have different ids for alternate APIs

Reply:
While true that it's a subset, we'll have to use the CurrencyId in the dispatchable functions/rpcs, so we'll have to deal with two very similar types, and convert between them when necessary. Plus we'd be duplicating information, introducing the possibility for mismatches. You really think that's worth it?